Core/GDCore/IDE/Events/EventsPositionFinder.h

```diff
@@ -0,0 +1,50 @@
+#ifndef EventsPositionFinder_H
+#define EventsPositionFinder_H
+#include "GDCore/Events/EventsList.h"
+#include "GDCore/IDE/Events/ArbitraryEventsWorker.h"
+#include "GDCore/String.h"
+
+namespace gd {
+class BaseEvent;
+class EventsList;
+}  // namespace gd
+
+namespace gd {
+
+/**
+ * \brief Scans an event list to retrieve the position of a list of searched
+ * events when the events list is flattened.
+ *
+ * \ingroup IDE
+ */
+class GD_CORE_API EventsPositionFinder : public ArbitraryEventsWorker {
+ public:
+  EventsPositionFinder() {
+    index = 0;
+  };
+  virtual ~EventsPositionFinder();
+
+  /**
+   * Return the positions of all searched events
+   */
+  const std::vector<std::size_t> GetPositions() { return positions; }
```

Comment:
Return a const ref:
```suggestion
  const std::vector<std::size_t>& GetPositions() const { return positions; }
```

This won't change the usage in JS, but this will allow to avoid making a copy of this vector in memory in C++.

Also make the method const because it does not change anythign and this will be useful to someone else who wants to use your class as a const reference.